inst/include/distributions/functors/density_components_base.hpp

```diff
@@ -34,12 +35,20 @@ struct DensityComponentBase : public fims_model_object::FIMSObject<Type> {
   // Assigning each one its own ID is a way to keep track of
   // all the instances of the DensityComponentBase class.
  static uint32_t id_g; /**< global unique identifier for distribution modules */
- fims::Vector<Type> x; /**< input value of distribution function */
+ uint32_t observed_data_id_m = -999; /*!< id of observed data component*/
+ std::shared_ptr<fims_data_object::DataObject<Type>>
+      observed_values; /**< observed data*/
+ fims::Vector<Type > x; /**< input value of distribution function for proirs or random effects*/
  fims::Vector<Type> expected_values; /**< expected value of distribution function */
  fims::Vector<Type> lpdf_vec; /**< vector to record observation level negative log-likelihood values */
- std::string  lpdf_type; /**< string classifies the type of the negative log-likelihood; options are: prior, re, data */
+ std::string  input_type; /**< string classifies the type of the negative log-likelihood; options are: prior, re, data */
```

Comment:
Maybe specify the real options in the documentation: "priors", "random_effects", and "data"?

Reply:
Done.